src/shared/nuclearnet/tests/nuclearnet_proxy_parser.tests.ts

```diff
@@ -0,0 +1,170 @@
+import { NUClearNetPacket } from 'nuclearnet.js'
+
+import { Encoder } from '../nuclearnet_proxy_parser'
+import { Decoder } from '../nuclearnet_proxy_parser'
+import { TYPES } from '../nuclearnet_proxy_parser_socketio'
+import { Packet } from '../nuclearnet_proxy_parser_socketio'
+import { EventPacket } from '../nuclearnet_proxy_parser_socketio'
+
+describe('NUClearNetProxyParser', () => {
+
+  let e: Encoder
+  let d: Decoder
+
+  beforeEach(() => {
+    e = new Encoder()
+    d = new Decoder()
+  })
+
+  it('Roundtrips regular socket.io packets', () => {
+
+    const packets: Packet[] = [
+      {
+        nsp: '/',
+        type: TYPES.CONNECT,
+      },
+      {
+        nsp: '/',
+        type: TYPES.DISCONNECT,
+      },
+      {
+        nsp: '/',
+        type: TYPES.ACK,
+        data: [],
+        id: Math.floor(Math.random() * 1000),
```

Comment:
`SeededRandom` please 😄  + create a `randomPacketId()` helper function below the `beforeEach`.